src/hotspot/cpu/x86/x86.ad

```diff
@@ -1814,6 +1814,11 @@ bool Matcher::supports_vector_variable_shifts(void) {
   return (UseAVX >= 2);
 }
 
+bool Matcher::supports_vector_variable_rotates(VectorNode *n) {
```

Comment:
Please, turn it into a capability check (just `bool Matcher::supports_vector_variable_rotates(void)`, no argument).
There's no need to wrap `Matcher::match_rule_supported_vector` since you already check it:
```
(in(2)->is_Con() && !Matcher::supports_vector_variable_rotates(this)) ||
!Matcher::match_rule_supported_vector(Op_RotateLeftV, vlen, bt)) {
```

Otherwise, looks good.

Reply:
Done. Turned the capability check  into `bool Matcher::supports_vector_variable_rotates(void)`. On x86_64, it returns `true` directly now.